subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -0,0 +1,296 @@
+
+
+/**
+ * @brief Local function used to error check snprintf outputs.
+ *
+ * @param[in] ret Return value of snpritnf.
+ * @param[in] size Max size of what would have been the final string.
+ * @param[in] entry Name of the function it was called from.
+ */
+static inline void report_snprintf_err(u32_t ret, u32_t size, char *entry)
+{
+	if (ret >= size) {
+		LOG_ERR("Unable to fit formated string into to allocate"
+			"memory for %s", entry);
+	} else if (ret < 0) {
+		LOG_ERR("Output error for %s was encountered with return value"
+			" %d", entry, ret);
+	}
+}
+
+
+/**
+ * @brief Constructs the get topic for accepted and rejeceted AWS jobs.
+ *
+ * @param[in] client_id Client ID of the current connected MQTT session.
+ * @param[out] topic Outputs the constructed mqtt_topic into the given buffer.
+ */
+static void construct_get_topic(u8_t *client_id, struct mqtt_topic *topic)
+{
+	char get_topic[GET_TOPIC_LEN + 1];
+	int ret = snprintf(get_topic,
+			   GET_TOPIC_LEN,
+			   GET_TOPIC_TEMPLATE,
+			   client_id,
+			   "#");
+	report_snprintf_err(ret, GET_TOPIC_LEN, "get_topic");
+
+	topic->topic.utf8 = get_topic;
+	topic->topic.size = ret + 1;
+	topic->qos = MQTT_QOS_1_AT_LEAST_ONCE;
+	return;
+}
+
+
+int aws_jobs_subscribe_get(struct mqtt_client *client)
+{
+	struct mqtt_topic subscribe_topic = construct_get_topic(
+						client->client_id.utf8
+						);
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&subscribe_topic,
+		.list_count = 1,
+		.message_id = SUBSCRIBE_GET
+	};
+	LOG_DBG("Subscribe: %s\n", subscribe_topic.topic.utf8);
+
+	return mqtt_subscribe(client, &subscription_list);
+}
+
+
+/**
+ * @brief Constructs the notify-next topic for reciving AWS IoT jobs.
+ *
+ * @param[in] client_id Client ID of the current connected MQTT session.
+ * @param[out] topic Outputs the constructed mqtt_topic into the given buffer.
+ */
+static void construct_notify_next_topic(u8_t *client_id,
+					struct mqtt_topic *topic)
+{
+	char notify_next_topic[NOTIFY_NEXT_TOPIC_MAX_LEN  + 1];
+	int ret = snprintf(notify_next_topic,
+			   NOTIFY_NEXT_TOPIC_MAX_LEN,
+			   NOTIFY_NEXT_TOPIC_TEMPLATE,
+			   client_id);
+	report_snprintf_err(ret,
+			    NOTIFY_NEXT_TOPIC_MAX_LEN,
+			    "notify_next_topic");
+
+	topic->topic.utf8 = notify_next_topic;
+	topic->topic.size = ret + 1;
+	topic->qos = MQTT_QOS_1_AT_LEAST_ONCE;
+	return;
+}
+
+
+int aws_jobs_subscribe_notify_next(struct mqtt_client *client)
+{
+	struct mqtt_topic subscribe_topic;
+	construct_notify_next_topic(client->client_id.utf8, &subscribe_topic);
+
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&subscribe_topic,
+		.list_count = 1,
+		.message_id = SUBSCRIBE_NOTIFY_NEXT
+	};
+	LOG_DBG("Subscribe: %s\n", subscribe_topic.topic.utf8);
+
+	return mqtt_subscribe(client, &subscription_list);
+}
+
+
+/**
+ * @brief Constructs the notifyt topic for reciving AWS IoT jobs.
```

Comment:
Typo, `notifyt`

Reply:
Wow I wonder how my spellchecker didn't catch that one tnx :) will fix it